tests/lib/acpi/src/main.c

```diff
@@ -17,14 +33,44 @@ ZTEST(acpi, test_mcfg_table)
 	zassert_not_null(mcfg, "Failed to get MCFG table");
 }
 
-ZTEST(acpi, test_irq_routing_table)
+ZTEST(acpi, test_dev_enum)
 {
-	static ACPI_PCI_ROUTING_TABLE irq_prt_table[CONFIG_ACPI_MAX_PRT_ENTRY];
-	int status;
+	struct acpi_dev *dev;
+	ACPI_RESOURCE *res_lst;
+	int ret;
+
+	Z_TEST_SKIP_IFNDEF(ACPI_DT_HAS_HID(DT_NODELABEL(pcie0)));
```

Comment:
The `Z_TEST_SKIP_IFNDEF()` macro is AFAIK only intended to be used for Kconfig options, which are either undefined (if they're set to `n`) or defined (if they're set to `y`). However here you always get something that's defined (either `0` or `1`). I think what this really should be checking for is the same condition that the PCIe code is using to require the ACPI properties or not, i.e. CONFIG_PCIE_PRT:
```c
	Z_TEST_SKIP_IFNDEF(CONFIG_PCIE_PRT);
```

Reply:
> Nothing to do with PRT. Updated by define APCI_TEST_DEV macro if HID property exist. I observed Z_TEST_SKIP_IFNDEF skip test case if node property does not exist.

Yep. I just checked the implementation, and this is using `COND_CODE_1()` which will only evaluate to "true" if the given macro is both defined *and* has the value `1`. In that sense the `SKIP_IFNDEF()` name is slightly misleading, since it's not just looking for whether the macro is defined or not but also its value.